### PR TITLE
Fixes

### PR DESCRIPTION
**Pull request by dshanske**
Discovered that the code as previously written was not associating tokens properly based on cnditionals for the Indieweb plugin. This should fix that and it is locked to either returning the root URL or the author URL.

Also, for web-signin, to ensure security, unique URL properties for user accounts are now enforced. URLs can now be seen when viewing all users.

Also added IndieAuth to the site health check and changed the language in the success/failure scripting. 

Need to do some additional scope hardening, but this should fix the problem I discovered this week until I can.

### CODE REVIEW

File: includes/class-indieauth-admin.php [hunk 1]
```diff
@@ -11,6 +11,96 @@ public function __construct() {
...
+		}
+		return $val;
+	}
+
+	public function add_user_url_column( $column ) {
+		$column['user_url'] = __( 'Website', 'indieauth' );
+		return $column;
+	}
+
+			/**
```
Comment (pfefferle) [inline includes/class-indieauth-admin.php]: indents!

File: includes/class-indieauth-admin.php [hunk 2]
```diff
@@ -62,9 +151,9 @@ public function admin_init() {
 		add_settings_field( 'indieauth_general_settings', __( 'IndieAuth Settings', 'indieauth' ), array( $this, 'general_settings' ), 'general', 'default' );
 	}
 
-	/**
-	 * Add IndieAuth options to the WordPress general settings page.
-	 */
+			/**
```
Comment (pfefferle) [inline includes/class-indieauth-admin.php]: indents!

File: includes/class-indieauth-admin.php [hunk 3]
```diff
@@ -75,9 +164,9 @@ public function general_settings() {
 		printf( __( 'Based on your feedback and to improve the user experience, we decided to move the settings to a separate <a href="%1$s">settings-page</a>.', 'indieauth' ), $path ); // phpcs:ignore
 	}
 
-	/**
-	 * Add admin menu entry
-	 */
+			/**
```
Comment (pfefferle) [inline includes/class-indieauth-admin.php]: indents!

File: includes/class-indieauth-admin.php [hunk 4]
```diff
@@ -123,23 +209,35 @@ public function settings_page() {
...
 			$json = json_decode( wp_remote_retrieve_body( $response ) );
-			set_query_var( 'authdiag_message', $json->message );
+			return $json->message;
 		} else {
-			set_query_var( 'authdiag_message', 'Fail' );
+			return false;
 		}
+	}
 
+			/**
```
Comment (pfefferle) [inline includes/class-indieauth-admin.php]: indents!

File: indieauth.php [hunk 5]
```diff
@@ -3,7 +3,7 @@
  * Plugin Name: IndieAuth
  * Plugin URI: https://github.com/indieweb/wordpress-indieauth/
  * Description: IndieAuth is a way to allow users to use their own domain to sign into other websites and services
- * Version: 3.3.2
+ * Version: 3.3.3
```
Comment (pfefferle) [inline indieauth.php]: This sounds greater than a "patch"?

File: readme.txt [hunk 6]
```diff
@@ -125,6 +125,10 @@ Some hosting providers filter this out using mod_security. For one user, they ne
 
 == Upgrade Notice ==
 
+= 3.3.3 =
+
+Due to the possibility of someone setting the url in their user profile to the same as another account, you will no longer be able to save the exact same url into two accounts
```
Comment (pfefferle) [inline readme.txt]: what if two already use the same domain?

File: indieauth.php (hunk 5 above)
Comment (dshanske) [reply indieauth.php]: You think it should bump to 3.4? It doesn't seem to bring any new features...although I suppose the unique URL thing is a feature.

File: readme.txt (hunk 6 above)
Comment (dshanske) [reply readme.txt]: The system would discard the second one and give anyone access to the first. But, you think it should be flagged and addressed?

**Comment by dshanske** [comment]
I will fix the indents. The biggest reason for these changes was the possibility that the system as it existed would return a token for a user other than the one who logged in.

File: indieauth.php (hunk 5 above)
Comment (pfefferle) [reply indieauth.php]: and it breaks backwards compatibility... if only a little ;)

Comment (dshanske) [reply indieauth.php]: Issuing corrections now.

**Comment by dshanske** [comment]
@pfefferle Bumped to 3.4.0

**Review by pfefferle — COMMENTED** [review]
Shouldn't it throw an error if two users use the same URL? Otherwise it still is a security issue!

File: readme.txt (hunk 6 above)
Comment (pfefferle) [reply readme.txt]: Shouldn't it throw an error if two users use the same URL? Otherwise it still is a security issue!

**Comment by dshanske** [comment]
Right now, it just silently refuses to save. I have to figure out how to generate an error on saving. I don't see anything in the filter to allow that.

**Comment by pfefferle** [comment]
Not while saving... When someone tries to login with an URL, that another person also uses, you said, the first one is used, but it should throw an error! Something like "Not possible to identify unique urser" or something like that... Otherwise it does not fix the issue. At least not, until all have changed their URLs...

**Comment by dshanske** [comment]
I just can't figure out how to throw that error. I don't see a way to surface it. I may have to dig deeper into the code.

**Comment by pfefferle** [comment]
Instead of the "Approve" page?

**Comment by dshanske** [comment]
You mean, check during the process the plugin controls? Interesting. That could work.

**Comment by dshanske** [comment]
@pfefferle This adds a few things. If the count of returns in searching for a URL is more than 1 it will now fail. And it does a check on the user list page for dupes.

**Comment by dshanske** [comment]
@pfefferle This ready to go?

**Review by pfefferle — APPROVED** [review]

**Pull request merged**